analyzers/src/SonarAnalyzer.Common/Rules/BackslashShouldBeAvoidedInAspNetRoutesBase.cs

```diff
@@ -0,0 +1,78 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class BackslashShouldBeAvoidedInAspNetRoutesBase<TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
```

Comment:
I would suggest the following:
* Have a single class for all route-specific rules, which
* does the extraction of the route once and calls all route rules from there (the single rules do not derive from SonarDiagnosticAnalyzer, with
* The route extraction being based on the `ArgumentTracker` from [this branch](https://github.com/SonarSource/sonar-dotnet/compare/master...Martin/S2077_PetaPoco2). The description should be or`ed depending on the availability of the attributes.

Reply:
Unlike [S6931](https://sonarsource.github.io/rspec/#/rspec/S6931/csharp), for example, this rule targets any route template, whether it is in a controller or not. For example, it also targets route templates defined in methods like `MapAreaControllerRoute`, `MapFallbackToAreaController`, etc., that can be defined anywhere (in `Program.cs`, in `Startup.cs`, in externalized `WebApplication` builder methods, ...).

Because of the larger scope, and the fact that this rule doesn't need to check whether we are in a controller or not, specifically for this rule, I don't see the added value of introducing an indirection level.

Instead, @mary-georgiou-sonarsource may find the approach valuable for S6931, and if so, we could evaluate ex post the opportunity of factorizing S6930 under the new "routes rule aggregator".

WDYT?